File: typescript/resistor-color-duo/resistor-color-duo.ts

export function decodedValue(colors: string[]): number{
  const config = new Map<string, number>([
    ['black', 0],
    ['brown', 1],
    ['red', 2],
    ['orange', 3],
    ['yellow', 4],
    ['green', 5],
    ['blue', 6],
    ['violet', 7],
    ['grey', 8],
    ['white', 9],
  ]);
  const x = `${config.get(colors[0])}${config.get(colors[1])}`
  const y: number = +x;
  return y;
}

File: elixir/resistor-color-duo/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  def code(:black), do: 0
  def code(:brown), do: 1
  def code(:red), do: 2
  def code(:orange), do: 3
  def code(:yellow), do: 4
  def code(:green), do: 5
  def code(:blue), do: 6
  def code(:violet), do: 7
  def code(:grey), do: 8
  def code(:white), do: 9

  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value([first, second | _]) do
    # Integer.undigits([code(first), code(second)])
    code(first) * 10 + code(second)
  end
end
